src/databricks/labs/ucx/workspace_access/generic.py

```diff
@@ -189,6 +186,26 @@ def _safe_get_permissions(self, object_type: str, object_id: str) -> iam.ObjectP
             else:
                 raise RetryableError() from e
 
+    @retried(on=[RetryableError])
+    def _safe_update_permissions(
```

Comment:
This handles errors now, but how does this ensure all permissions have been applied successfully?

Reply:
If no error is returned, can we assume the permissions have been applied?

Reply:
Apparently not? Please consult with @renardeinside, as there are seemingly cases where the function returns without error, but the permission is not available? I am just making the point that before it looped over presence of permissions, not that it runs without exceptions.

Reply:
Yes, indeed, there is now `_inflight_check` to check if the permissions have been applied as expected in `_applier_task` for each type

Reply:
There is an edge case in platform issue where grants are not available even though they return 200. [ ES ticket](https://databricks.atlassian.net/browse/ES-908737). This [PR](https://github.com/databrickslabs/ucx/pull/512) should solve this.
In general, when the API returns 200 the permission should be applied. If it's not then it should be reported as platform bug via ES ticket. What are the scenarios that we see the permissions are not being applied even though there is not exception?

Reply:
Apparently, [the API can be inconsistent](https://github.com/databrickslabs/ucx/blob/50301a6884d77bd244820c4f30a6a66038d03bce/src/databricks/labs/ucx/workspace_access/secrets.py#L84-L85), I don't know if the problem has been raised already to the platform

Reply:
Got it, we should fix that but this should also be reported to the platform.